src/reflect/scala/reflect/internal/Reporting.scala

```diff
@@ -90,13 +90,21 @@ abstract class Reporter {
   private[this] var _warningCount = 0
 
   // sbt source compatibility
-  type Severity = Reporter.Severity
+  final type Severity = Reporter.Severity
   @uncheckedStable final def INFO: Severity    = Reporter.INFO
   @uncheckedStable final def WARNING: Severity = Reporter.WARNING
   @uncheckedStable final def ERROR: Severity   = Reporter.ERROR
 
+  // TODO: rename to `doReport`, remove the `force` parameter.
+  // Note: `force` is ignored. It used to mean: if `!force`, the reporter may skip INFO messages.
+  // If `force`, INFO messages were always printed. Now, INFO messages are always printed.
   protected def info0(pos: Position, msg: String, severity: Severity, force: Boolean): Unit
 
+  // 0: count and display
+  // 1: count only, don't display
+  // 2: don't count, don't display
+  def filter(pos: Position, msg: String, severity: Severity): Int = 0
```

Comment:
Still on the fence about how to document, whether to use an enum. It's not an internal Rex Kerr counter thing, it's API. And there are three states, as when one speaks of the "tri-state area", everyone knows you mean New York and specifically Manhattan, no one ever goes out to 1 and 2 except to visit relatives.